designs/NoSQL_note.md

```diff
@@ -0,0 +1,83 @@
+# NoSQL
+
+Due to the technical limitations of setting up SQL, we decide to take the NoSQL approach by utilizing DynamoDB table on AWS, which has a trivial entry.
+
+## Tables
+
+### Customers:
+ - id: `string`
+ - email: `string`
+ - address: `string`
+ - payment_method: `TODO`
+ - phone_number: `string`
+ - order_ids: `List[string]`
+
+#### Example:
+ - id: Thomas
+ - email: xxxxx@gmail.com
+ - address: Unit 123 23th street, Vancouver,BC
+ - payment_method: `TODO`
+ - phone_number: 236863xxxx
+ - order_ids: [123321, 1234321, 5321321, ...]
+
+ ### Orders
+ - id: `string`
+ - customer_id: `string`
+ - restaurant_id: `string`
+ - content : `list[string]`
+ - extra_charge: `string`
+ - time: `time`
+ - driver_id: `string`
+ - status: `string`
+
+ #### Example:
+ - id: 1234567
+ - customer_id: 3213534
+ - restaurant_id: wd234eqedre32
+ - content: [dish_name: fry_chicken, unit price: $5, amount: 3]
+ - extra_charge: $5
+ - time: Jan 1st, 2023
+ - driver_id: 21e32edqw
+ - status: delivered
+
+ ### Restaurant
+ - restaurant_id: `string`
+ - restaurant_name: `string`
+ - restaurant_description: `string`
+ - restaurant_address: `string`
+ - restaurant_phone_number: `string`
+ - schedule: `time`
```

Comment:
Just leave as a TODO for now: time means a timestamp only. The actual schedule can be relatively complex.
```suggestion
 - schedule: `TODO`
```